### DISCUSSION TITLE
Make this project more useful for others

### DISCUSSION THREAD

**Discussion by fralken** [versat/cntlm#80] [Ideas]
Hi @versat, this project has evolved a lot since the original on Sourceforge, so it can be a good idea to make it more useful for others. For example I implemented a release workflow in my [fork](https://github.com/fralken/cntlm) to publish binaries.
Another idea could be to make the project more visible by creating a github organisation and transferring the repo to it, and advertising it in the old cntlm's forum on Sourceforge. What do you think?

**Comment by versat** [comment]
Yes, that sounds great. Transferring the project to an organization would be really useful, I currently have no idea how that works but we will find it out 🙂
Great that you had time to create a release workflow 👍

**Reply by fralken** [reply]
Yes I wanted to play with github actions just to learn how to implement them, and since the release workflow was missing and was a requested feature, I created it. Macos binaries are still missing but I'm planning to add them, both for x86 and arm. When ready I'll PR it..

As for organisations, there is a free plan, I can create an organisation, add members, and add or transfer repos. I already did one for another projects some time ago.

I have a question on your workflows: some of them are scheduled to run regularly like
```
  schedule:
    - cron: '0 7 * * 4'
```
is this really necessary? since nothing changes it looks redundant and wasting resources.

**Reply by versat** [reply]
Sorry for the late answer, I thought I already answered.
It would be great to transfer the repo to an organization, if you want to create and manage that I fully support that :)
Yes, some jobs are run regularly (once a week), and sometimes the static analyzers have new rules and detect new bugs. And it is also good to know that everything still works. But it is not really necessary. The scan is really quick, I do not think that too much resources are wasted, but I would be also ok with disabling it.

**Comment by fralken** [comment]
Hello @versat, hope you're doing well. I was thinking that this project has evolved a lot since the fork from Evengard.
Would it be a good idea to make it independent so that for example it appears in search queries?
This is possible via [support's virtual agent](https://support.github.com/?q=extract+fork). A fork can be "detached" or "extracted" from main repo. I think "extract" is our case since it becomes independent but still keeping children forks.
What do you think?

**Reply by jschwartzenberg** [reply]
I think that detaching it would only make it harder to find?

I see two things to do to reach more users:
1. Do a release with a version number
2. Get the new version downstreamed in the various GNU/Linux distributions

**Reply by fralken** [reply]
Well, a forked repository has some limitations, for what I now it is not shown in searches on Github, that is, you can find Evengard's repo and from it go to versat's one, but you cannot find it directly. Another thing, contributors are not shown in the home page, not a big deal, but still...
As for the release:

1. two years ago I made a Github action to publish a release, it's available in my fork. It is not polished, so I didn't PR it to versat's yet. When I find some spare time I'll try and improve it.
2. I admit I have no idea how to downstream to the distributions, such as Homebrew for mac, Apt and so on. Suggestions are welcome.

**Reply by versat** [reply]
@fralken Thank you for the link. I have requested to detach this fork and bring subforks with this repo (the common way and what makes sense here too).

A release would be good, I agree.

Do you have any idea how to add some (semi) automated tests? Using some (fake/test) proxies or so to verify that common functionality still works.

Regards
Sebastian / versat

**Reply by fralken** [reply]
It would be useful to have some automated tests but unfortunately I don't how to mock a ntlm server.

**Reply by jschwartzenberg** [reply]
cntlm's feature set goes beyond ntlm authentication nowadays. It is already useful to handle PAC files for applications that do not deal with them for instance. Tests with a proxy server that doesn't require authentication would already be useful

Note that when I tested IPv6 support, I could not use an ntlm proxy server either. I used a proxy server that doesn't require authentication, but was available over IPv6.
